exercises/practice/sieve/.approaches/introduction.md

```diff
@@ -0,0 +1,80 @@
+# Introduction
+
+The key to this exercise is to keep track of:
+- A list of numbers.
+- Their status of possibly being prime.
+
+## General Guidance
+
+To solve this exercise, it is necessary to choose one or more appropriate data structures to store numbers and status, then decide the best way to scan through them.
+
+There are many ways to implement the code, and the three broad approaches listed below are not sharply separated.
+
+## Approach: Using nested loops
+
+```python
+def primes(number):
+    not_prime = []
+    prime = []
+    
+    for item in range(2, number+1):
+        if item not in not_prime:
+            prime.append(item) 
+            for element in range(item*item, number+1, item):
+                not_prime.append(element)
+    
+    return prime
+```
+
+The theme here is nested, explicit `for` loops to move through ranges, testing validity as we go.
+
+For details and another example see [`nested-loops`][approaches-nested].
+
+## Approach: Using set operations
+
+```python
+def primes(number):
+    not_prime = set()
+    primes = []
+
+    for num in range(2, number+1):
+        if num not in not_prime:
+            primes.append(num)
+            not_prime.update(range (num*num, number+1, num))
+
+    return primes
+```
+
+In this group, the code uses the special features of the Python [`set`][sets] to improve efficiency.
+
+For details and other examples see [`set-operations`][approaches-sets].
+
+## Approach: Using complex or nested comprehensions
+
+```python
+def primes(limit):
+    return [number for number in range(2, limit + 1) if 
+                 all(number % divisor != 0 for divisor in range(2, number))]
+- For bit arrays, there is the [`bitarray`][bitarray] package and [`bitstring.BitArray()`][bitstring].
+- For arrays of booleans, we could use the NumPy package: `np.ones((number,), dtype=np.bool_)` will create a pre-dimensioned array of `True`.
+
```

Comment:
```suggestion

```python
def primes(limit):
    return [number for number in range(2, limit + 1) if 
                 all(number % divisor != 0 for divisor in range(2, number))]
```  


- For bit arrays, there is the [`bitarray`][bitarray] package and [`bitstring.BitArray()`][bitstring].
- For arrays of booleans, we could use the NumPy package: `np.ones((number,), dtype=np.bool_)` will create a pre-dimensioned array of `True`.

```